### PR TITLE
Hard hat/helmet attachments

### PR DESCRIPTION
**Pull request by Termineitor244**
#### Summary
Content "Adds hard hats/helmets attachments"

#### Purpose of change
Continuation of #58020 and part of #58025.
Many helmets or hard hats can have attachments and modifications, it's time for it to become reality in cataclysm too.

Examples of planned add-ons for hard hats:
[Sun shields](https://www.amazon.com/MSA-697410-Plastic-Shield-Non-Slotted/dp/B000WE809W)
[Face Shield and Mounting Bracket](https://www.amazon.com/Defender-Safety-Polycarbonate-Mounting-Bracket/dp/B08S5X6Z51) 
[Shield Visor](https://www.amazon.com/Shinmax-Adults-Accessory-Adjustable-Removable/dp/B08P2YDV92/)
[Ear Muffs](https://www.amazon.com/3M-XSeries-X2P3E-CapMount-Earmuffs/dp/B00CPCHDQA)
[Ballistic Mandible Guard](https://www.hcctac.com/collections/helmet-accessories/products/caiman-ballistic-mandible-guard?variant=34962231820442)
[Counterweight](https://www.uspatriottactical.com/agilite-detachable-universal-counterweight-pouch/agi_8280.html#start=2) [Pouch](https://es.aliexpress.com/item/32847466320.html)

#### Describe the solution
Adds many new attachments for hard hats helmets, as well as the pockets that will accommodate these new items in the helmets/hard hats.

I also modified the hooded hard hat to now use a sun shield, and added a neck protector attachment in place of the hooded hard hat.

PROTECTION/COVERAGE VALUES ARE WIP UNTIL I GO AROUND TESTING THEM

#### Describe alternatives you've considered
To not add them?

#### Testing
TO BE DONE.

#### Additional context
Thanks @bombasticSlacks for the help with making these new ablative pockets to work!

That and adding some spawns/recipes, as well as adding the new pockets to the rest of the helmets.

### CODE REVIEW

**Comment by bombasticSlacks** [comment]
Well if you've already got the json done I can try to do the infrastructure tomorrow if you would like?

**Comment by Termineitor244** [comment]
> Well if you've already got the json done I can try to do the infrastructure tomorrow if you would like?
> 
> I think I have an idea for how to do it in an easy way.

I would be very grateful if you did!

**Comment by TheShadowFerret** [comment]
Are any of these going to be craftable? Otherwise, non-master craftsmen need a repairs_like.

**Comment by Termineitor244** [comment]
> Are any of these going to be craftable? Otherwise, non-master craftsmen need a repairs_like.

At the very least the pouch will be craftable, the face shields too if I can base myself around some similar recipe, and the earmuffs maybe...

But anyways, I had forgotten about repairs_like, thanks for reminding me, I will add some after getting out of work.

**Comment by vonschtirlitz** [comment]
Oh man love this addition, I'll see if I can add something or contribute irl data!

Just playing super detail attentive devil's advocate here, most if not all civilian hard hats aren't really able to take specialty military attachments (counterweights, mandibles, etc). The ones I've seen can take earpro, and then smaller faceshields for tree work and the like. Ballistic helmets usually use proprietary interfaces to attach more involved items like mandibles and face shields, which do not work across different brands of helmets ([example 1](https://www.teamwendy.com/products/helmets-accessories/parts-accessories/exfil-ballistic-mandible), [example 2](https://www.galvion.com/collections/mandible-guards/products/caiman-ballistic-mandible-guard)), much less civilian hard hats.

Are we supposing that there is now one magic helmet attachment standard for civillian and military helmets? Or is it perhaps worth restricting some attachments to certain helmets, thereby adding some choice consideration, perhaps if you wanted a lighter, more modular helmet, versus a plain helmet that has more ballistic protection.

Either way, this looks super exciting and I'll try to contribute something as well! 😋

**Comment by Termineitor244** [comment]
> Oh man love this addition, I'll see if I can add something or contribute irl data!

Any one of those would be helpful! Thanks!

> Just playing super detail attentive devil's advocate here, most if not all civilian hard hats aren't really able to take specialty military attachments (counterweights, mandibles, etc). The ones I've seen can take earpro, and then smaller faceshields for tree work and the like. Ballistic helmets usually use proprietary interfaces to attach more involved items like mandibles and face shields, which do not work across different brands of helmets ([example 1](https://www.teamwendy.com/products/helmets-accessories/parts-accessories/exfil-ballistic-mandible), [example 2](https://www.galvion.com/collections/mandible-guards/products/caiman-ballistic-mandible-guard)), much less civilian hard hats.

The face shield example I gave above in the description is, in fact, a face shield made for hard hats! With the mandible attachment I'm going more for the "You could kind of adapt one to use straps for hard hats, it wouldn't be ideal, but it would allow you to wear it without a military helmet". The counterweight has 2 links on the description, in one of those links the pouch is just hanging from some wire wrapped around your helmet, so why not use something similar for most helmets/hard hats?

> Are we supposing that there is now one magic helmet attachment standard for civillian and military helmets? Or is it perhaps worth restricting some attachments to certain helmets, thereby adding some choice consideration, perhaps if you wanted a lighter, more modular helmet, versus a plain helmet that has more ballistic protection.

Well, there are going to be some compromises for the sake of simplicity, like assuming all military helmets have the interfaces needed for these kind of military attachments, but I definitely want to limit the possible attachments depending on the helmet, mainly because of shape, but without limiting too much if I can help it, specially since I want the attachments to cover most of the sublocations that the helmets will not cover after they undergo an update to he sublimb system.

> Either way, this looks super exciting and I'll try to contribute something as well! 😋 

Thank you! I'm glad you are on board with the idea!

**Comment by bombasticSlacks** [comment]
> Are we supposing that there is now one magic helmet attachment standard for civillian and military helmets? 

We do this for gun mods currently, no differentiation between Picatinny rails, mlok, Russian side mounts etc.  

For a lot of this stuff it's getting into "too detailed" territory. This isn't Escape From Tarkov you shouldnt need a spread sheet to match face shields with military helmets IMO at least.

 Another way to think about this is realistically there will probably only be 3-4 military helmets. High cut w. Attachment points, ears covering w. Attachment points, maybe a high end or sci Fi helm made of better materials, and then 1-2 helmets without attachment points. So there are only realistically going to be 2-3 helmets most people will use so going nuts on differentiation doesn't make much sense.

**Comment by vonschtirlitz** [comment]
> I'm going more for the "You could kind of adapt one to use straps for hard hats, it wouldn't be ideal, but it would allow you to wear it without a military helmet"

Ok yeah I see what you mean. I should also probably take into account the crazy mad ~~scientist~~ engineer survivor who can probably duct tape stuff to his helmet to attach it instead of using fancy "pockets" and attachment points. This would probably also make improvised applique armor (like football helmet front cage) pretty viable and accessible early game.

> We do this for gun mods currently, no differentiation between Picatinny rails, mlok, Russian side mounts etc.

I'm completely on the same page and I agree with this approach, but it would feel weird to be able to freely click in some ballistic face shield or future AR goggles onto some random hard hat or bike helmet you found. Perhaps if you were to modify the helmet to add additional proper slots or something it would make sense, versus a military helmet coming standard with those. I haven't touched CDDA in a few months recently, but I remember there being either a mod or some other system that allowed you to add additional weapon modification slots to weapons that didn't have them to begin with. Maybe something along the lines of that.

**Comment by bombasticSlacks** [comment]
> I'm completely on the same page and I agree with this approach, but it would feel weird to be able to freely click in some ballistic face shield or future AR goggles onto some random hard hat or bike helmet you found.

if you look at what is done in this PR there are new mount points actually defined. So a bike helmet just wouldn't have slots. A helmet that covers the ears wouldn't have ear slots but could have a visor slot. ETC. This system is very opt-in not opt-out when it comes to including items in it 😆 

I'm not 100% convinced on the mandibles for hard hats but at worst it could be revisited in the future IMO once more options have these mount points.

**Comment by Termineitor244** [comment]
I plan on changing the hooded hard hat to be one that uses a sun guard, and make a new attachment for a nape protector (To replace the original hooded hard hat).

The new hard hat with sun guard will be the same as the normal version, but with a sunglasses effect and somewhat less possible attachments (Because of its size mainly).

**Comment by toweroglass** [comment]
> Perhaps if you were to modify the helmet to add additional proper slots or something it would make sense, versus a military helmet coming standard with those. I haven't touched CDDA in a few months recently, but I remember there being either a mod or some other system that allowed you to add additional weapon modification slots to weapons that didn't have them to begin with. Maybe something along the lines of that.

I like this solution a lot. would seem more in line with the way we mod old guns to make them more "tacticool". a kit to attach mandibles to a hard hat could just require some plastic chunks and the mandible to be used as a tool for reference. Obviously i over simplified the recipe but something like that.

Early game I always dream i could combine a mining helmet and a hooded hard hat. Seems like this kind of modularity shouldn't be difficult. some of the mix and match situations might not make sense irl, but in game it stands to reason you would be making the most with what you come across.

File: data/json/items/armor/hats.json [hunk 1]
```diff
@@ -524,8 +576,8 @@
     "id": "hat_hard_hooded",
     "type": "ARMOR",
     "category": "armor",
-    "name": { "str": "hooded hard hat" },
-    "description": "A hard plastic hat worn in construction or industrial sites, with a cloth tucked in under it to serve as a makeshift hood.  Makes it a bit more comfortable to wear, and better protects the neck from sun and rain.",
+    "name": { "str": "summer hard hat" },
+    "description": "A hard plastic hat worn in construction or industrial sites, with a sun shield around it to serve as sun protection.  Cover your eyes of sun glare, and better protects you from the heat of a summer day.",
```
Comment (TheShadowFerret) [inline data/json/items/armor/hats.json:580]:
```suggestion
    "description": "A hard plastic hat worn in construction or industrial sites, with a sun shield around it to serve as sun protection.  Shields your eyes from sun glare, and better protects you from the heat of a summer day.",
```

File: data/json/items/armor/head_attachments.json [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        "specifically_covers": [ "head_nape", "head_ear_r", "head_ear_l" ]
+      }
+    ]
+  },
+  {
+    "id": "kevlar_nape_protector",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": { "str": "Kevlar nape protector" },
+    "description": "A custom made Kevlar strengthened nape protector intended to be attached on a hard hat or helmet to protect your nape against cuts and slashing attacks.",
```
Comment (TheShadowFerret) [inline data/json/items/armor/head_attachments.json:217]:
```suggestion
    "description": "A custom-made Kevlar strengthened nape protector intended to be attached on a hard hat or helmet to protect your nape against cuts and slashing attacks.",
```

**Comment by Termineitor244** [comment]
This branch got broken, so I moved my changed to #58749.

**Pull request closed without merging**